src/Repository/SongRepository.php

```diff
@@ -91,9 +105,15 @@ public function getRandomByArtist(
         $sql   = (AmpConfig::get('catalog_disable'))
             ? "SELECT DISTINCT `song`.`id` FROM `song` LEFT JOIN `artist_map` ON `artist_map`.`object_id` = `song`.`id` AND `artist_map`.`object_type` = 'song' LEFT JOIN `catalog` ON `catalog`.`id` = `song`.`catalog` WHERE `artist_map`.`artist_id` = ? AND `artist_map`.`object_type` = 'song' AND `catalog`.`enabled` = '1' "
             : "SELECT DISTINCT `artist_map`.`object_id` AS `id` FROM `artist_map` WHERE `artist_map`.`artist_id` = ? AND `artist_map`.`object_type` = 'song' ";
+
+        if (AmpConfig::get('catalog_filter') && !empty(Core::get_global('user'))) {
+            $sql .= "AND" . Catalog::get_user_filter('song', Core::get_global('user')->id) . " ";
+        }
+
         $sql .= "ORDER BY RAND()";
 
         $db_results = Dba::read($sql, array($artist->getId()));
+//        debug_event(self::class, 'getRandomByArtist ' . $sql, 5);
```

Comment:
make sure the pull is set to allow edits cause i'll basically hit it and reupload things that are out of place

Reply:
The checkbox on the right side is checked.

Those commented out debug_events appear in a lot of places,. btw.